### PR TITLE
BUG: Fix wrong khash method definition

### LINKED ISSUE #20722: BUG: Overflow encountered when using .loc with UInt64Index
#### Code Sample, a copy-pastable example if possible

```python
>>> import pandas as pd
>>> import numpy as np
>>> s = pd.Series([1, 2], index=[np.iinfo('uint64').max - 1, np.iinfo('uint64').max])
>>> s[np.iinfo('uint64').max]
2
>>> s.loc[np.iinfo('uint64').max]

...
OverflowError: Python int too large to convert to C long
During handling of the above exception, another exception occurred:
KeyError
...

>>> s.index
UInt64Index([18446744073709551614, 18446744073709551615], dtype='uint64')
```
#### Problem description

`loc` leads to an overflow even with consistent dtype (`uint64` for the index of the Series and the value pass to `loc`).

#### Expected Output

``` python
>>> s.loc[np.iinfo('uint64').max]
2
```

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Linux
OS-release: 4.13.0-38-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: 3.3.2
pip: 9.0.1
setuptools: 38.5.1
Cython: 0.27.3
numpy: 1.14.0
scipy: 1.0.0
pyarrow: None
xarray: 0.10.1
IPython: 6.2.1
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.1.2
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```

</details>

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #20722
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -784,3 +784,16 @@ def convert_nested_indexer(indexer_type, keys):
...
+        # GH20722
+        s = pd.Series(
+            [1, 2],
+            index=[
+                np.iinfo('uint64').max - 1,
+                np.iinfo('uint64').max
+            ]
+        )
+
+        assert 1 == s.loc[np.iinfo('uint64').max - 1]
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: can you construct the expected frame and user assert_series_equal

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -784,3 +784,16 @@ def convert_nested_indexer(indexer_type, keys):
             index=pd.MultiIndex.from_product(keys))
 
         tm.assert_series_equal(result, expected)
+
+    def test_loc_uint64(self):
+        # GH20722
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: can you add a 1 line comment about what this is about

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew in bugs / indexing

**Comment by Licht-T** [comment]
Thanks @jreback, fixed!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green. I pushed some cleaning of the test.

**Comment by jreback** [comment]
thanks @Licht-T

**Pull request merged**